assets/src/components/carousel/carousel-nav.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Navigation dot.
+ *
+ * @param {Object} props Component props.
+ * @param {boolean} props.isCurrent Whether the dot is currently selected.
+ * @param {boolean} props.isHighlighted Whether the current item is highlighted.
+ * @param {string} props.id An HTML ID.
+ * @param {string} props.label Button label.
+ * @param {string} props.namespace CSS namespace.
+ * @param {Function} props.onClick Click callback.
+ */
+function Dot( { id, isCurrent, isHighlighted, label, namespace, onClick } ) {
+	return (
+		<Button
+			className={
+				[
+					`${ namespace }__nav-dot-button`,
+					isCurrent ? `${ namespace }__nav-dot-button--current` : '',
+					isHighlighted ? `${ namespace }__nav-dot-button--active` : '',
+				]
+					.filter( ( item ) => item )
+					.join( ' ' )
+			}
+			id={ id }
+			onClick={ onClick }
+			aria-label={ label }
+		>
+			<span className="components-visually-hidden">
+				{ label }
+			</span>
+			{ isCurrent && (
+				<span className="components-visually-hidden">
+					{ __( '(Current slide)', 'amp' ) }
```

Comment:
I suppose you meant to say `item` here.

```suggestion
					{ __( '(Current item)', 'amp' ) }
```